### ISSUE TITLE
Forking doesn't work on Kovan or other networks

### ISSUE THREAD

**Issue by dmihal** [nomiclabs/hardhat#1038]
I just ran into the following issue while trying to fork Kovan using an Alchemy API

```
  Error: Invalid RpcBlockWithTransactions or null
      at codec.decode.fold (/Users/dmihal/keystone/defi-777/contracts/node_modules/hardhat/src/internal/hardhat-network/jsonrpc/types.ts:8:11)
      at Left.fold (/Users/dmihal/keystone/defi-777/contracts/node_modules/io-ts/node_modules/fp-ts/lib/Either.js:113:16)
      at Object.decode (/Users/dmihal/keystone/defi-777/contracts/node_modules/hardhat/src/internal/hardhat-network/jsonrpc/types.ts:6:30)
      at JsonRpcClient._perform (/Users/dmihal/keystone/defi-777/contracts/node_modules/hardhat/src/internal/hardhat-network/jsonrpc/client.ts:230:27)
      at process._tickCallback (internal/process/next_tick.js:68:7)
```

The issue is also reported by @PatrickAlphaC here:
https://ethereum.stackexchange.com/questions/89951/hardhat-deploy-script-works-on-testnet-but-not-on-forked-hardhat-network

**Comment by fvictorio** [comment]
This is the same issue that's preventing xdai from being forked, the kovan network returns different block objects than mainnet and rinkeby. It might be an easy fix, but we need to look into it in more detail to see if that fix doesn't causes other issues.

Just out of curiosity, why are you forking Kovan instead of the mainnet?

**Comment by PatrickAlphaC** [comment]
@fvictorio sometimes we have stuff on kovan we want to test. ie, I have a bunch of [Chainlink price feed](https://docs.chain.link/docs/get-the-latest-price) addresses on Kovan, and I want to make sure those addresses are responding correctly.

**Comment by fabioDMFerreira** [comment]
I also got stuck on this issue. I'd like to test my DApp on Kovan before deploying to the mainnet. 

Is this issue on your roadmap?

**Comment by fvictorio** [comment]
> Is this issue on your near future roadmap?

Yes, we are looking into this right now. If there aren't any major issues, we'll have it working soonish.

**Comment by PatrickAlphaC** [comment]
Also, other EVM compatible blockchains would be great to fork. I just tried this on matic and xdai with no success.

**Comment by fvictorio** [comment]
Hey everyone, we just released `hardhat@2.0.9` that supports forking Kovan. Please let us know if something doesn't work!
